toolchain/sem_ir/typed_insts.h

```diff
@@ -47,7 +47,7 @@ namespace Carbon::SemIR {
 
 // Used for the type of patterns that do not match a fixed type.
 //
-// TODO: Annotate as a builtin.
+// This is a singleton instruction.
```

Comment:
Would it be possible to add something like
```
static constexpr InstId SingletonId = SingletonInstId<AutoType>;
```
so that we can say `SemIR::AutoType::SingletonId` instead of `SemIR::SingletonInstInd<SemIR::AutoType>`?

Reply:
Done, with some refactoring to make it the favored path.